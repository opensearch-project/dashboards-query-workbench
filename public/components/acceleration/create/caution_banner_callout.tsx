/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiCallOut } from '@elastic/eui';
import React from 'react';

export const CautionBannerCallout = () => {
  return (
    <EuiCallOut title="Considerations for data indexing" color="warning" iconType="iInCircle">
      <p>
        Warning about not indexing personal or sensitive data, something about the cost of indexing.
      </p>
    </EuiCallOut>
  );
};
