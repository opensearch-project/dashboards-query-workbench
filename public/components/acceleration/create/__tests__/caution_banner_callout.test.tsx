/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { waitFor } from '@testing-library/dom';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { CautionBannerCallout } from '../caution_banner_callout';

describe('Acceleration callout', () => {
  configure({ adapter: new Adapter() });

  it('renders acceleration flyout callout', async () => {
    const wrapper = mount((<CautionBannerCallout />) as React.ReactElement<any>);
    wrapper.update();
    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
