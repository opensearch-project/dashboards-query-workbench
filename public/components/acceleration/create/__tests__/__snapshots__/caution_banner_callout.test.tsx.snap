// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Acceleration callout renders acceleration flyout callout 1`] = `
<div
  className="euiCallOut euiCallOut--warning"
>
  <div
    className="euiCallOutHeader"
  >
    <svg
      aria-hidden={true}
      className="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiCallOutHeader__icon"
      focusable="false"
      height={16}
      role="img"
      style={null}
      viewBox="0 0 16 16"
      width={16}
      xmlns="http://www.w3.org/2000/svg"
    />
    <span
      className="euiCallOutHeader__title"
    >
      Considerations for data indexing
    </span>
  </div>
  <div
    className="euiText euiText--small"
  >
    <div
      className="euiTextColor euiTextColor--default"
    >
      <p>
        Warning about not indexing personal or sensitive data, something about the cost of indexing.
      </p>
    </div>
  </div>
</div>
`;
